File: backend/app/schemas/school.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class SchoolCreate(BaseModel):
    name: str
    email: str | None = None
    website: str | None = None
    address: str | None = None


class SchoolUpdate(BaseModel):
    name: str | None = None
    email: str | None = None
    website: str | None = None
    address: str | None = None


class SchoolResponse(BaseModel):
    id: UUID
    name: str
    email: str | None = None
    website: str | None = None
    address: str | None = None
    is_active: bool
    created_at: datetime

File: backend-java/src/main/java/com/easylearn/entity/School.java
package com.easylearn.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "schools")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class School {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    private String email;

    private String website;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}
